caffe2/serialize/inline_container.cc

```diff
@@ -641,7 +641,9 @@ void PyTorchStreamWriter::setup(const string& file_name) {
     const std::string dir_name = parentdir(file_name);
     if(!dir_name.empty()) {
       struct stat st;
-      bool dir_exists = (stat(dir_name.c_str(), &st) == 0 && (st.st_mode & S_IFDIR));
+      bool dir_exists =
+          (stat(dir_name.c_str(), &st) == 0 && (st.st_mode & S_IFDIR)) ||
+          (dir_name.length() == 2 && dir_name.back() == ':');
```

Comment:
- Guard it under `#ifdef WIN32` (as on Linux/Mac it's fine to write to a file named `C:`)
- Add check `isalpha(dir_name.front()` (unless `_:` is also considered a drive name)
- What about `lpt:`, `com:`, etc

Reply:
I replaced this check with function calls from the filesystem library, which shouldn't need guards and are platform independent